app/services/ValidatingPageBuilder.scala

```diff
@@ -182,21 +190,31 @@ class ValidatingPageBuilder @Inject() (pageBuilder: PageBuilder){
         }
     }
 
-  private def checkForSequencePageReuse(connections: List[PageVertex])(implicit stanzaMap: Map[String, Stanza]): List[GuidanceError] = {
-    val vertexMap = connections.map(pv => (pv.id, pv)).toMap
-    val mainFlow: List[String] = mainFlowPageIds(vertexMap)
-    val sequencePageIds = for{
-      pv <- connections.filterNot(_.flows.isEmpty)              // Sequences
+  private def checkAllFlowsHaveUniqueTerminationPage(vertices: List[PageVertex])(implicit stanzaMap: Map[String, Stanza]): List[GuidanceError] = {
+    val vertexMap = vertices.map(pv => (pv.id, pv)).toMap
```

Comment:
vertexMap and mainFlow are also invoked in the method checkForSequencePageReuse. You could define them in the main validation method and pass them into validation methods as required.

Reply:
done